### ISSUE TITLE
OptionsPattern inspection should not trigger within SyntaxInformation

### ISSUE THREAD

**Issue by szhorvat** [WolframResearch/codeinspector#3]
The following triggers the OptionsPattern inspection:

```
CodeInspectSummarize[
 "SyntaxInformation[f] = {\"ArgumentsPattern\" -> {_., OptionsPattern[]}}"
 ]
```

![image](https://user-images.githubusercontent.com/1212871/80234967-4c632a00-8659-11ea-81d5-ea48d0ccb481.png)

It should not. This is standard usage within `SyntaxInfomation`.

**Comment by szhorvat** [comment]
Could `OptionsPattern[]` ever reasonably occur at a level deeper than 1 in "ArgumentsPattern"? I am wondering about the best way to detect that it is part of a `SyntaxInformation`. Do we need to scan for `Set[SyntaxInformation[...], ...]` at arbitrary levels up in the AST, or only at a fixed location relative to the occurrence of `OptionsPattern[]`?

**Comment by bostick** [comment]
Thank you for filing this. This was on an internal list from an earlier email, but it is good to keep track.

**Comment by bostick** [comment]
This is fixed in the latest sources, and also requires latest sources from CodeParser as well
